src/commonMain/kotlin/org/kosat/Incremental.kt

```diff
@@ -0,0 +1,87 @@
+package org.kosat
+
+import kotlin.math.abs
+
+class Incremental(initClauses: MutableList<MutableList<Int>>, initNumber: Int = 0) : CDCL(initClauses, initNumber) {
+    // assumptions for incremental sat-solver
+    private var assumptions: List<Int> = emptyList()
+
+    fun solveWithAssumptions(currentAssumptions: List<Int> = emptyList()): List<Int>? {
+        assumptions = currentAssumptions
+        val result = solve()
+        assumptions.forEach {
+            if (getStatus(it) == VarStatus.FALSE) {
+                assumptions = emptyList()
+                return null
+            }
+        }
+        assumptions = emptyList()
+        return result
+    }
+
+    override fun getNextVariable(level: Int): Int {
+        return if (level > assumptions.size) {
+            vsids()
+        } else {
+            return assumptions[level - 1]
+        }
+    }
+
+
+    override fun solve(): List<Int>? {
+        removeUselessClauses()
+
+        // extreme cases
+        if (clauses.isEmpty()) return emptyList()
+        if (clauses.any { it.size == 0 }) return null
+        if (clauses.any { it.all { lit -> getStatus(lit) == VarStatus.FALSE } }) return null
+
+        countScore()
+
+        buildWatchers()
```

Comment:
Be careful not to "build watchers" twice (for "old" clauses). I do not *yet* fully understand your implementation (maybe this situation is already well-handled), so I just warn you that this `solve` method could be called multiple times (since this is IncrementalSolver).